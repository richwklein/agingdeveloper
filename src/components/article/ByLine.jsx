import React from "react";
import {Box, Typography} from "@mui/material";
import InternalLink from "../common/InternalLink";
import PropTypes from "prop-types";

const ByLine = ({authorName, authorSlug, publishedDate}) => {
  return (
    <Box>
      <Typography variant="subtitle2"
        sx={{
          display: "inline-block",
          backgroundColor: "primary.dark",
          color: "primary.contrastText",
          marginRight: .25,
        }}>
        <InternalLink to={`/author/${authorSlug}`}
          sx={{
            display: "inline-block",
            backgroundColor: "inherit",
            color: "inherit",
            textDecoration: "underline",
            padding: .75,
          }}>
          {authorName}
        </InternalLink>
      </Typography>
      <Typography variant="subtitle2" sx={{
        display: "inline-block",
        padding: .75,
      }}>
        {publishedDate}
      </Typography>
    </Box>
  );
};

ByLine.propTypes = {
  authorName: PropTypes.string.isRequired,
  authorSlug: PropTypes.string.isRequired,
  publishedDate: PropTypes.string.isRequired,
};

export default ByLine;
