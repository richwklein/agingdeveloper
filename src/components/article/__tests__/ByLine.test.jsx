import React from "react";
import {cleanup, render} from "@testing-library/react";
import ByLine from "../ByLine";

describe("ByLine", () => {
  afterEach(cleanup);

  const renderComponent = () => render(
      <ByLine
        author={{name: "Georgia Smith", slug: "georgia-smith"}}
        date="November 1st, 2023" />,
  );

  test("that it renders correctly.", () => {
    const {container} = renderComponent();
    expect(container.firstChild).toMatchSnapshot();
  });

  test("that it has a link to the authors page.", () => {
    const {container} = renderComponent();
    expect(container.firstChild).toContainHTML("href=\"/author/georgia-smith\"");
  });
});
