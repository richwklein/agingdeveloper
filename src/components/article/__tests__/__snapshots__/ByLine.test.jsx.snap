// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ByLine that it renders correctly. 1`] = `
<div
  class="MuiBox-root css-0"
>
  <h6
    class="MuiTypography-root MuiTypography-subtitle2 css-1ab5it3-MuiTypography-root"
  >
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-qb75xw-MuiTypography-root-MuiLink-root"
      data-link="internal"
      href="/author/georgia-smith"
    >
      Georgia Smith
    </a>
  </h6>
  <h6
    class="MuiTypography-root MuiTypography-subtitle2 css-15yipx9-MuiTypography-root"
  >
    November 1st, 2023
  </h6>
</div>
`;
